### PR TITLE
If the user role changes to something different than what is stored in the session, update the session with the new role

### PR DESCRIPTION
**Pull request by jesseparent**
If the user role changes to something different than what is stored in the session, update the session with the new role
account for ids coming in as strings or numbers

### CODE REVIEW

File: controllers/api/user-routes.js [hunk 1]
```diff
@@ -116,12 +116,43 @@ router.put('/:id', withAuthAdmin, (req, res) => {
...
         const deleteUserSQL = `DELETE FROM session WHERE data LIKE '%"user_id":${req.params.id},%'`;
-        console.log(deleteUserSQL);
         sequelize.query(deleteUserSQL);
       }
+
+      const userRole = parseInt(req.body.role_id);
+      if (userRole === 1) { // User is set to developer role
+        // Find a session object for this user with the opposite role
+        const { QueryTypes } = require('sequelize');
+        const findSessionSQL = `SELECT * FROM session WHERE data LIKE '%"user_id":4,%"role_id":2%'`;
```
Comment (BradCassityDev) [inline controllers/api/user-routes.js]: Is the user_id hardcoded here?

File: controllers/api/user-routes.js [hunk 2]
```diff
@@ -116,12 +116,43 @@ router.put('/:id', withAuthAdmin, (req, res) => {
...
+              sessionData[0].data = sessionData[0].data.replace('"role_id":2,', '"role_id":1,');
+              const replaceRoleSQL = `UPDATE session SET data = '${sessionData[0].data}' WHERE sid='${sessionData[0].sid}'`;
+              sequelize.query(replaceRoleSQL);
+            }
+          });
+      }
+      else if (userRole === 2) { // user is set to admin role
+        // Find a session object for this user with the opposite role
+        const { QueryTypes } = require('sequelize');
+        const findSessionSQL = `SELECT * FROM session WHERE data LIKE '%"user_id":4,%"role_id":1%'`;
```
Comment (BradCassityDev) [inline controllers/api/user-routes.js]: Is the user_id hardcoded here?

File: controllers/api/user-routes.js (hunk 1 above)
Comment (jesseparent) [reply controllers/api/user-routes.js]: Changed to req.params.id

File: controllers/api/user-routes.js (hunk 2 above)
Comment (jesseparent) [reply controllers/api/user-routes.js]: Changed to req.params.id

**Review by BradCassityDev — APPROVED** [review]

**Pull request merged**